lib/project_types/extension/models/specification.rb

```diff
@@ -1,8 +1,34 @@
 module Extension
   module Models
-    Specification = Struct.new(:identifier, keyword_init: true) do
-      def handler_class_name
-        identifier.split('_').map(&:capitalize).join
+    class Specification
+      include SmartProperties
+
+      module Features
+        class Argo
+          include SmartProperties
+
+          property! :surface_area, converts: :to_str
+          property! :renderer_package_name, converts: :to_str
+          property! :git_template, converts: :to_str
+        end
+
+        def self.build(feature_set_attributes)
+          feature_set_attributes.each_with_object(OpenStruct.new) do |(identifier, feature_attributes), feature_set|
+            feature_set[identifier] = ShopifyCli::ResolveConstant
+              .call(identifier, namespace: Features)
+              .rescue { OpenStruct }
+              .then { |c| c.new(**feature_attributes) }
+              .unwrap { |error| raise error }
+          end
+        end
+      end
+
+      property! :identifier
+      property :graphql_identifier, converts: :to_str
```

Comment:
Could potentially be extracted into `messages` as this just influences CLI presentation and what `--type` takes as a value.